draft-chen-cfrg-vdaf-pine.md

```diff
@@ -259,40 +261,46 @@ derived from shares of the gradient.
-> meant to be used as a reference, not to fully specify things.
-
 | Parameter             | Type     | Description |
 |:----------------------|:---------|:------------|
-| alpha                 | XXX      | XXX         |
-| num_wr_checks         | XXX      | XXX         |
-| num_wr_successes      | XXX      | XXX         |
-| encoded_sq_norm_bound | XXX      | XXX         |
-| wr_check_bound        | XXX      | XXX         |
-| num_bits_for_sq_norm  | XXX      | XXX         |
-| num_bits_for_wr_check | XXX      | XXX         |
-| chunk_len             | XXX      | XXX         |
+| alpha                 | float    | Parameter in wraparound check that determines the ZK error. The higher `alpha` is, the lower ZK error is. |
+| num_wr_checks         | int      | Number of wraparound checks to run. |
+| num_wr_successes      | int      | Minimum number of wraparound checks that a Client must pass. |
+| encoded_sq_norm_bound | Field    | The square of `l2_norm_bound` encoded into a field element. |
+| wr_check_bound        | Field    | The bound of the range check in wraparound checks. |
+| num_bits_for_sq_norm  | int      | Number of bits to encode the squared L2-norm. |
+| num_bits_for_wr_check | int      | Number of bits to encode the range check in each wraparound check. |
+| chunk_length          | int      | Chunk length used in FLP proof. |
 {: #pine-flp-param title="Operational parameters of the PINE FLP."}
 
 ## Measurement Encoding
 
-> XXX Give an overview of `encode_gradient()` and `encode_wr_checks()`. No need
-> to go into full detail yet. The goal is to give the reader an intuition for
-> what the circuit will do, including the bit checks, norm computation, and
-> wraparound checks. Be sure to emphasize that the wraparound test results are
-> not transmitted: instead the Aggregators derive shares of the tests results
-> by running the wraparound tests themselves.
-
-### Encoding of Floating Point Numbers into Field Elements
+The measurement encoding is done in two stages:
+* `PineValid.encode_gradient(self, measurement: list[float]) -> list[Field]`
+  allows a Client to encode each floating point number in its gradient into a
+  field element {{float-to-field}}. Then the Client encodes the results for
+  L2-norm check {{l2-norm-check}}, by computing the bit representation of the
+  squared L2-norm, modulo `q`, of its encoded gradient. The result of this step
+  allows Aggregators to check the squared L2-norm of the Client's gradient,
+  modulo `q`, falls in the desired range of `[0, encoded_sq_norm_bound]`.
+* `PineValid.encode_wr_checks(self, encoded_gradient: list[Field],
+  wr_joint_rand_xof: Xof) -> list[Field]` allows the Client to encode the
+  results of wraparound checks {{wr-check}}, based on the `encoded_gradient`
+  from the previous step, and the random vector generated from an initialized
+  `Xof` in `wr_joint_rand_xof. The result of this step, along with the encoded
```

Comment:
Instead of speaking of an XOF here, we should only speak of the seed used to derive the random vectors via the XOF.

Reply:
isn't the XOF composed of the seed and domain separation tag? Mentioning XOF here also connects to our reference implementation?